crates/context/config/src/client/starknet.rs

```diff
@@ -0,0 +1,208 @@
+#![allow(clippy::exhaustive_structs, reason = "TODO: Allowed until reviewed")]
+
+use core::str::FromStr;
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use serde::{Deserialize, Serialize};
+use starknet_core::types::{BlockId, BlockTag, Felt, FunctionCall};
+use starknet_core::utils::get_selector_from_name;
+use starknet_providers::jsonrpc::HttpTransport;
+use starknet_providers::{JsonRpcClient, Provider, Url};
+use thiserror::Error;
+
+use super::{Operation, Transport, TransportRequest};
+
+#[derive(Copy, Clone, Debug, Deserialize, Serialize)]
+#[serde(try_from = "serde_creds::Credentials")]
+pub struct Credentials {
+    pub account_id: Felt,
+    pub public_key: Felt,
+    pub secret_key: Felt,
+}
+
+mod serde_creds {
+    use core::str::FromStr;
+
+    use serde::{Deserialize, Serialize};
+    use starknet_crypto::Felt;
+    use starknet_types_core::felt::FromStrError;
+
+    #[derive(Debug, Deserialize, Serialize)]
+    pub struct Credentials {
+        secret_key: String,
+        public_key: String,
+        account_id: String,
+    }
+
+    impl TryFrom<Credentials> for super::Credentials {
+        type Error = FromStrError;
+
+        fn try_from(creds: Credentials) -> Result<Self, Self::Error> {
+            let public_key_felt = Felt::from_str(&creds.public_key)?;
+            let secret_key_felt = Felt::from_str(&creds.secret_key)?;
+            let account_id_felt = Felt::from_str(&creds.account_id)?;
+
+            Ok(Self {
+                account_id: account_id_felt,
+                public_key: public_key_felt,
+                secret_key: secret_key_felt,
+            })
```

Comment:
the config file is user-editable, so we still need to verify that the public key was derived from the secret